### PR TITLE
nautilus: rgw/pubsub: fix records/event json format to match documentation

### PR DESCRIPTION
**Pull request by smithfarm**
backport tracker: https://tracker.ceph.com/issues/43168

---

backport of https://github.com/ceph/ceph/pull/31926
parent tracker: https://tracker.ceph.com/issues/43056

this backport was staged using ceph-backport.sh version 15.0.0.6950
find the latest version at https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

**Comment by smithfarm** [comment]
NOTE: Since this is the first post-14.2.5 release note, I took advantage of this PR to clean up `PendingReleaseNotes`.

**Comment by yuriw** [comment]
@smithfarm needs rebase

--- pr 32221 --- pulling https://github.com/smithfarm/ceph.git branch wip-43168-nautilus
remote: Enumerating objects: 22, done.
remote: Counting objects: 100% (22/22), done.
remote: Total 25 (delta 22), reused 22 (delta 22), pack-reused 3
Unpacking objects: 100% (25/25), done.
From https://github.com/smithfarm/ceph
 * branch                  wip-43168-nautilus -> FETCH_HEAD
Auto-merging PendingReleaseNotes
CONFLICT (content): Merge conflict in PendingReleaseNotes
Automatic merge failed; fix conflicts and then commit the result.
Traceback (most recent call last):
  File "/home/yuriw/wip_master/src/script/build-integration-branch", line 67, in <module>
    assert not r
AssertionError

**Comment by yuriw** [comment]
[wip-yuri4-testing-2020-01-27-1545-nautilus](https://trello.com/c/6Y0veAm2/817-wip-yuri4-testing-2020-01-27-1545-nautilus)

File: PendingReleaseNotes [hunk 1]
```diff
@@ -1,123 +1,4 @@
...
-  do the same but only including heartbeats initiated by the specified OSD.
-
-* New OSD daemon command dump_recovery_reservations which reveals the
-  recovery locks held (in_progress) and waiting in priority queues.
-
-* New OSD daemon command dump_scrub_reservations which reveals the
-  scrub reservations that are held for local (primary) and remote (replica) PGs.
-
-14.2.6
+14.2.7
```
Comment (cbodley) [inline PendingReleaseNotes]: was it the intent to remove all prior release notes in this backport?

**Comment by yuriw** [comment]
@smithfarm this passed tests and ready to merge after resolving @cbodley comment

File: PendingReleaseNotes (hunk 1 above)
Comment (smithfarm) [reply PendingReleaseNotes]: @cbodley Yes, that was intentional. Did I do it wrong?

The notes in `PendingReleaseNotes` for prior releases are no longer pending. They are cut in stone over at  https://github.com/ceph/ceph/blob/master/doc/releases/nautilus.rst

`PendingReleaseNotes` is just a scratch pad. Often what happens is someone will paste a release note onto the end (or insert at the beginning) and when we have the notes for previous releases still in there, it's not clear to which release the note applies.

Comment (cbodley) [reply PendingReleaseNotes]: ok! the changes just looked unrelated to this specific pr, but i can see the logic in it

**Pull request merged**